sycl/CMakeLists.txt

```diff
@@ -532,6 +532,11 @@ if("hip" IN_LIST SYCL_ENABLE_BACKENDS)
   list(APPEND SYCL_TOOLCHAIN_DEPLOY_COMPONENTS ur_adapter_hip)
 endif()
 
+if(INSTALL_DEVICE_CONFIG_FILE)
+  add_dependencies(sycl-toolchain DeviceConfigFile)
```

Comment:
can you give some background on why we need to install this? thx

Reply:
I'm still testing things, so I might change some things, but I want to support the new `device_config_file_consistency` test. It uses the `DeviceConfigFile.hpp`, and that file includes `DeviceConfigFile.inc`, which is generated by tablegen. On CI from what I understand the e2e tests are invoked by using the packed install files from the build step and only runs CMake on the `sycl/test-e2e` subfolder. So since we probably don't want to build tablegen and invoke other LLVM cmake files when running the e2e tests, I install the `DeviceConfigFile.inc` it in the build step to pass it to the e2e tests. Also note that this test must be an e2e test as it queries the device it is running the test on, so it can't be moved to `sycl/test`.

Reply:
sorry so what tools/files are required to generate that hpp file. Is it just `llvm-tablegen` and `DeviceConfigFile.inc`?

Reply:
Yea just tablegen and `DeviceConfigFile.td` are needed for the hpp file

Reply:
Also just for some more background I was aiming so that DeviceConfigFile.hpp is not installed by default because outside of testing, this file is not needed for a SYCL distribution, it is only used in the compiler.